python/cudf/cudf/core/frame.py

```diff
@@ -27,9 +28,83 @@ def _gather(self, gather_map):
         result = self.__class__._from_table(
             libcudfxx.gather(self, as_column(gather_map))
         )
+        result._update_index_name(self)
         result._copy_categories(self)
         return result
 
+    def dropna(self, how="any", subset=None, thresh=None):
```

Comment:
Can we move the axis kwarg into here as well so that the entire definition can be dropped from DataFrame / MultiIndex and it can share those implementations?

Reply:
@kkraus14 In that case have to move _drop_na_columns over here as well

Reply:
Yup. I think `drop_na_columns` can be implemented at the `Frame` level

Reply:
Also still need shim methods for `Index.dropna()` and `Series.dropna()` that call `super.dropna()`.

`Series.dropna()` should not expose a `subset` parameter.